tests/main/gadget-update-pc/task.yaml

```diff
@@ -0,0 +1,164 @@
+summary: Exercise a gadget update on a PC
+
+environment:
+    BLOB_DIR: $(pwd)/fake-store-blobdir
+    # snap-id of 'pc' gadget snap
+    PC_SNAP_ID: UqFziVZDHLSyO3TqSWgNBoAdHbLI4dAH
+    START_REVISION: 1000
+
+systems: [ubuntu-core-*]
+
+prepare: |
+    if ! test -d /snap/pc; then
```

Comment:
Could we add a sanity check that ensures this runs at least on ubuntu-core-16-64 and ubuntu-core-18-64? I think we have something like this in other tests, the idea is something like:
```sh
if  ! test -d /snap/pc && [ $SPREAD_SYSTEM = ubuntu-core-18.04-64 ]; then 
    echo the test must run here something is fishy
    exit 1;
fi
```
Just to ensure we don't accidentally not run the test but we don't notice because of the exit 0 (I know its a bit of a long shoot but …)

Reply:
Won't that be a problem when running the suite manually on an external, eg. RPi device?

Reply:
Added a check to fail on non-external backends. Systems whitelist already limits the test only to `ubuntu-core*` ones.

Reply:
The check for the test keys should take care of this, the external backends usually won't have the test keys.

Reply:
Ok, let me move the check for TRUST_TEST_KEYS.